okio/src/main/java/okio/AsyncTimeout.java

```diff
@@ -310,50 +302,13 @@ public Watchdog() {
     public void run() {
       while (true) {
         try {
-          AsyncTimeout timedOut = awaitTimeout();
-
-          // Didn't find a node to interrupt. Try again.
-          if (timedOut == null) continue;
-
+          final AsyncTimeout timedOut = queue.take();
           // Close the timed out node.
           timedOut.timedOut();
         } catch (InterruptedException ignored) {
         }
+        Thread.yield();
```

Comment:
So my thought was, if there are multiple items immediately on the queue to take, this could be executing for awhile with the loop. If we yield out the thread we would block other threads less.